### DISCUSSION TITLE
Best Approach for Adding an Order Tracking System to a Custom Packaging Website

### DISCUSSION THREAD

**Discussion by FredrickMichel** [community/community#146471] [Programming Help]
### Select Topic Area

Question

### Body

Hello GitHub Community,

I'm currently working on a project for my website, [To Go Packaging](https://customtakeoutboxes.com/) which offers custom packaging services. As I expand, I'd like to implement an order tracking feature to allow customers to check the status of their packaging orders directly on the site.

I’m exploring two potential solutions:

Building a custom tracking system: This would allow customers to enter their order number on my website and view real-time shipping information. Ideally, I’d like to integrate with popular shipping carriers via APIs, but I’m not sure which is the best method to go about this.

Redirecting customers to the carrier’s website: This would involve sending customers to the shipping company’s tracking page with pre filled details but I’m wondering if this might reduce the overall user experience.

I’d love to hear from anyone with experience implementing order tracking on eCommerce websites. Specifically, I’m looking for advice on:

Best practices for integrating shipping API services.
Potential challenges or pitfalls in building a custom tracking system.
Recommendations for tools or libraries that could help streamline the process.
Thank you for your time and expertise!

**Comment by mecodeatlas** [comment]
Thanks for posting in the GitHub Community, @FredrickMichel!

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the [GitHub product Discussions](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/about-discussions). This question should be in the Programming Help category. I've gone ahead and moved it for you. Good luck!
